### PR TITLE
Remove oqc installation and test

### PR DESCRIPTION
**Pull request by rmoyard**
**Context:**
https://pypi.org/project/types-pkg-resources/#history was completely removed, pytket < 1.30.0 depends on it and the oqc client depends on pytket.
**Description of the Change:**
Remove the oqc client installation until they have changed pytket dependency.

### CODE REVIEW

**Review by dime10 — APPROVED** [review]
Will the tests be skipped automatically?

**Comment by rmoyard** [comment]
Yes I think so (tested locally)

File: .github/workflows/check-catalyst.yaml [hunk 1]
```diff
@@ -409,7 +409,7 @@ jobs:
         # It can't be in requirements.txt as that will break
         # macOS requirements.txt
         python3 -m pip install cuda-quantum==0.6.0
-        python3 -m pip install oqc-qcaas-client
+        # python3 -m pip install oqc-qcaas-client
```
Comment (erick-xanadu) [inline .github/workflows/check-catalyst.yaml:412]:
```suggestion
        python3 -m pip install oqc-qcaas-client --no-deps
```

**Comment by rmoyard** [comment]
Released: https://pypi.org/project/oqc-qcaas-client/3.9.1/

**Pull request closed without merging**